src/plugins/socket.io.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const Tags = require('opentracing').Tags
+
+const reservedEvents = [
+  'error',
+  'connect',
+  'disconnect',
+  'disconnecting',
+  'newListener',
+  'removeListener',
+  'ping',
+  'pong'
+]
+
+function createPatchEmit (tracer, config) {
+  return function wrapEmit (emit) {
+    return function emitWithTrace (eventName, callback) {
+      if ((config.omitReserved && reservedEvents.includes(eventName)) ||
+          (config.omitEvents && config.omitEvents.includes(eventName))) {
+        return emit.apply(this, arguments)
+      }
+
+      const scope = tracer.scope()
+      const childOf = scope.active()
+
+      const tags = {
+        [Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_CLIENT,
+        'component': 'socket.io',
+        'namespace': this.name,
+        'path': this.server._path,
+        'clients.count': this.server.eio.clientsCount,
+        'events.count': this._eventsCount
+      }
+
+      const span = tracer.startSpan('emit ' + eventName, { childOf, tags: tags })
+
+      try {
+        scope.bind(this, span)
```

Comment:
What is this intended to do?

Reply:
Sorry, leftovers from earlier experimentation. Removed